azavea/franklin
# Add search filtering for search extension

### Overview

This adds the `STAC` search API extension support for filtering based on query parameters and a `POST` request with a json body.

There are a few other changes as well.

The biggest change I think is the collapse of the subprojects into a single project. The thinking here is that this makes development a little easier and since any library-like functionality should be moved into `stac4s` we don't lose anything in portability or reduced artifact size. Reducing to a single project makes it easier to identify packages that should be removed if they are no longer necessary.

Everything is changed to use `bloop` and run commands through that - this aligns with our giter8 template at Azavea.

Additionally, the `DAO` from RF is imported here. At least until that functionality gets moved out into its own artifact the functionality is copied over here. Still works out nicely.

`./scripts/test` now does way more things - checking formatting, running scapegoat, etc.

### Notes

Right now the tests confirm that the search service doesn't explode, but I think ideally we have some more specific tests to ensure that the tests function as we expect with some fixtures -- I opened an [issue](https://github.com/azavea/franklin/issues/31) for that.

A few generators are copied over from `stac4s` - I opened an [issue](https://github.com/azavea/stac4s/issues/2) over there to publish those. Once that happens we can work on this [issue](https://github.com/azavea/franklin/issues/30).

### Testing
- Run `./scripts/setup`
- If you haven't imported data, import the Panama dataset
- Start the server: `./scripts/server`
- Make some requests to verify that filtering works. Here are some examples:
```
http :9090/stac/search datetime==2019-10-01T18:37:11Z/2019-12-01T18:37:11Z limit==2
echo '{"datetime":"2019-10-01T18:37:11Z/2019-12-01T18:37:11Z","items":[],"collections":[]}' | http :9090/stac/search
```

Closes #3
